### ISSUE TITLE
Suggestion : add OpenStreeMap as map engine

### ISSUE THREAD

**Issue by Spheerys** [osucomm/geotv#4]
I'm a contributor to OSM mapping and I prefer to use OSM for mapping activities : it's generaly more precise, and I don't like Google anymore :)
Maybe it's a big deal to implement OSM on GeoTV, but if it's not, that's my suggestion for a future release : make an option during GeoTV installation or on a system key to choose which mapper we want :)

**Comment by electrickite** [comment]
I don't have any fundamental problems using OSM. I'm personally not as experienced with their API, so Google is my go-to choice. Does OSM provide the same drawing features (polygons, etc) as the Google Maps API?

**Comment by Spheerys** [comment]
I think yes, but I'm not an expert with the OSM API.
In fact, I can also ask a friend to have a look on this point :)

**Comment by electrickite** [comment]
If OSM were implemented as a mapping provider, Google Maps should remain available as an option. Some locations have better coverage through Google. It's also my understanding that in some countries (notably China) using crowd-sourced mapping data like OSM is illegal.

**Comment by Spheerys** [comment]
Yes of course : we plan to try to implement OSM on GeoTV as an option next weeks.
At minimum a working principe.
In my aera, GMaps is very poor and I also do my best to avoid using Google tools :)

I think an option on TV parameters to choose between GMaps and OSM very nice.

**Comment by Spheerys** [comment]
What is the best way to develop the OSM support ? We  make another branch on our cloned repository and we replace the GMaps API calls ?

**Comment by electrickite** [comment]
Yes, working on a separate feature branch would be a good idea.

The best approach might be to re-write the point and area drawing interface using something like Leaflet.js. Then use OSM or Google to provide the tile sets.

Something similar to this: https://github.com/shramov/leaflet-plugins/blob/master/examples/google.html

**Comment by RichardHitier** [comment]
Hi, I'm currently working on the OSM rewriting. 
But I 'm still stuck on one point: allowMultiple is always set to 1 even if i set it to 'No' in the "input option" page.
 any clue appreciated :-)

thanks.

**Comment by Spheerys** [comment]
A little up because we are blocked with Richard.
But maybe we have to open a new issue for this specific problem ?

**Comment by Spheerys** [comment]
A pull request is on the way :)

**Comment by Spheerys** [comment]
Here it is : https://github.com/electrickite/geotv/pull/18
Thanks Richard !!

Once merged and upgraded on MODX, you have to change the key setting geotv.enable_osm to enable.
You can still use GMaps by setting it to disabled.

**Comment by Spheerys** [comment]
Up :)

**Comment by Spheerys** [comment]
Hello @electrickite :)

Can you please take a look to our job, and evently ask question or say something about the OSM option ?
